framework/src/outputs/Checkpoint.C

```diff
@@ -47,6 +47,11 @@ Checkpoint::validParams()
       "cp",
       "This will be appended to the file_base to create the directory name for checkpoint files.");
 
+  // Since it only makes sense to write checkpoints at the end of time steps,
+  // change the default value of execute_on to TIMESTEP_END
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum = {EXEC_TIMESTEP_END};
```

Comment:
```suggestion
  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
  exec_enum = {EXEC_TIMESTEP_END};
  params.suppressParameter<ExecFlagEnum>("execute_on");
```

Makes it no longer settable by a user